### PR TITLE
Add Referrals Page

### PR DESCRIPTION
**Pull request by dhaselhan**
* Add CRUD functions for Referrals
* Show Legacy ID on Card Dialog and Header
* Show Due Date on Card

### CODE REVIEW

**Review by trslater — APPROVED** [review]
LGTM 🚀

File: alcs-frontend/src/app/features/planning-review/referrals/create/create-planning-referral-dialog.component.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    @Inject(MAT_DIALOG_DATA)
+    public data: {
+      planningReviewUuid: string;
+      minReceivedDate: number;
+    },
+    private dialogRef: MatDialogRef<CreatePlanningReferralDialogComponent>,
+    private planningReferralService: PlanningReferralService,
+  ) {
+    this.minimumDate = new Date(this.data.minReceivedDate);
+    debugger;
```
Comment (mhuseinov) [inline alcs-frontend/src/app/features/planning-review/referrals/create/create-planning-referral-dialog.component.ts:42]: looks like a debug leftover

**Review by mhuseinov — APPROVED** [review]
👍

File: alcs-frontend/src/app/features/planning-review/referrals/create/create-planning-referral-dialog.component.ts (hunk 1 above)
Comment (dhaselhan) [reply alcs-frontend/src/app/features/planning-review/referrals/create/create-planning-referral-dialog.component.ts]: Good catch. removed

**Pull request merged**